### PR TITLE
Table sorting with Joins

### PR DESCRIPTION
**Pull request by patrickdent**
This implements sorting with join tables for everything except roles in the user#index page. sent you an email re: that issue.

This still requires that we create a system for choosing the primary author for each book. Once that is completed we can just change the logic in the case statement in the appropriate controllers.

Withholding tests on author sort until the above is completed.

### CODE REVIEW

**Comment by cherimarie** [comment]
In Keyword#show, on sort by author, the number of entries in the table doubles. See screenshots. Is this because of books with more than one author? 
screen shot 2015-06-03 at 10 33 35 am
screen shot 2015-06-03 at 10 33 45 am

Yes! When sorting a category that only has one book with more than one author, that is the only book that is duplicated.

**Comment by cherimarie** [comment]
screen shot 2015-06-03 at 10 37 25 am
screen shot 2015-06-03 at 10 37 34 am
This is where we really need the "main" author implemented, because that's the only author on whose name we want to sort.

**Comment by patrickdent** [comment]
In sort, yes.

This will be solved as soon as we implement some sort of primary author
scheme. The tables will then sort based that author's sort_by.

<3
p

**Comment by cherimarie** [comment]
Sort on title isn't working for me on any of the pages it's implemented on here.

File: app/controllers/keywords_controller.rb [hunk 1]
```diff
@@ -10,65 +10,69 @@ def index
...
+    end
+  end
 
   def show
     case params["sort"]
-    when "name"
+    when "cat_name"
       @books = @keyword.books.joins(:genre).includes(:authors)
-      .order(sort_column("title") + " " + sort_direction).paginate(:page => params[:page], :per_page => 50)
+      .order(sort_column("name") + " " + sort_direction).paginate(:page => params[:page], :per_page => 50)
```
Comment (cherimarie) [inline app/controllers/keywords_controller.rb]: need 'lower' here so it's case insensitive.

**Pull request merged**